src/App.tsx

```diff
@@ -14,11 +15,78 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[];
+};
+
+class App extends React.Component<unknown, State> {
+  state: State = {
+    selectedGoods: ['Jam'],
+  };
+
+  removeSelectedGoods = () => {
+    this.setState(() => ({
+      selectedGoods: [],
+    }));
+  };
+
+  addSelectedGood = (good: string) => (
+    this.setState(prevState => ({
+      selectedGoods: [...prevState.selectedGoods, good],
+    }))
```

Comment:
краще назвати currentState
```suggestion
    this.setState(currentState => ({
      selectedGoods: [... currentState.selectedGoods, good],
    }))
```